src/librustc_borrowck/borrowck/mir/gather_moves.rs

```diff
@@ -616,6 +616,16 @@ fn gather_moves<'a, 'tcx>(mir: &Mir<'tcx>, tcx: TyCtxt<'a, 'tcx, 'tcx>) -> MoveD
                         Rvalue::InlineAsm { .. } => {}
                     }
                 }
+                StatementKind::SetDiscriminant{ .. } => {
+                    // SCOTT FIXME
+                    // is this right?
+                    bb_ctxt.builder.create_move_path()
```

Comment:
I think you call just call `span_bug!(...)` here. Borrowck is operating before any discriminant operations are introduced.